### DISCUSSION TITLE
Is Betterfox good for someone having thousands of tabs opened (discarded) over multiple windows?

### DISCUSSION THREAD

**Discussion by matru** [yokoffing/Betterfox#331] [General]
My workflow consists of a lot tabs being opened, albeit discarded (unloaded), but there can be 200-400 tabs active at any point of time.

Would Betterfox work for me?

**Comment by yokoffing** [comment]
If Firefox works for you, then the Betterfox tweaks would also work for you.

also https://github.com/yokoffing/Betterfox/issues/302#issuecomment-2192006129

**Reply by ioc2e3** [reply]
> 如果 Firefox 適合您，那麼 Betterfox 調整也適合您。
> 
> 另見[#302 (評論)](https://github.com/yokoffing/Betterfox/issues/302#issuecomment-2192006129)

Thank you for your reply ...

.

**Comment by Feejus** [comment]
Dear Matru,

I just wanted to expand on this. The BetterFox user.js optimizes the standard Firefox by turning off things like Telemetry and reducing loading times. However, I must admit that having 200-400 tabs open is quite a lot to keep track of.

**Comment by matru** [comment]
@yokoffing thanks!

@Feejus very aware of it, but with systems I got in place, including treestyle tree tabs and various grouping methods, and various binds it's something I really find myself using productively, and since context switching is something I do every so often, permanent bookmarking does not work well in my workflow. The cost of this is lots of RAM usage, which is a big negative, but macOS handles things well, and incomparably so compared to linux and windows.

**Reply by yokoffing** [reply]
Firefox on macOS is almost like a different browser. Performance-wise, it works very well out of the box (for me).

Until Mozilla develops a more concerted effort towards memory management, I’d use Auto tab discard or similar extension for resource management, with that many tabs open.

**Reply by matru** [reply]
is there no real benefit in terms of performance then on macos by using Betterfox?

**Reply by philg-dev** [reply]
I'm using the "Simple Tab Groups" addon and I've got 800+ tabs open at all times with generally no big issues with RAM.

I have some specific tab groups configured to be automatically discarded (i.e. unload tab from memory, while not closing it), which works great for me.

In some cases when I have a bunch of really memory intensive tabs loaded in-memory, I use `about:processes` to manually discard the ones that use the most memory (again without closing them).

Another thing: If you happen to use a bunch of pinned tabs - those love to eat memory by default, because they will always be loaded into memory, even when restarting Firefox. This behavior can be changed to on-demand loading with:

```
browser.sessionstore.restore_pinned_tabs_on_demand = true
```

Maybe some of this helps.

**Reply by tweaksalot** [reply]
Thanks @philg-dev that was helpful especially the on-demand loading change

**Reply by matru** [reply]
@philg-dev thanks! I used to use STG long time ago, it's pretty great! Are you on a mac and if so how much ram do you have? Also do you have any preferred settings in regards to number of writes to disk/time or any other optimizations?

**Reply by philg-dev** [reply]
@matru I'm on Arch Linux with 64 GB of RAM installed. Firefox usually takes anywhere between 1GB and up to around 7~8 GB when I have a bunch of memory-intensive tabs open **and loaded**, e.g. several tabs of twitch, YouTube, LinkedIn, google docs, etc. The number of tabs in my STG groups doesn't really matter, it's been 800+ tabs and growing for a long time already. I usually have less than 30 tabs loaded in memory in my workflow.

I'm actually mostly on "vanilla" Firefox settings in terms of performance. I don't use betterfox at the moment. I'm just following it right now because I plan on adopting settings from betterfox step-by-step at some point.

Literally the only noteworthy thing that comes to mind is what I've already mentioned above:

```js
browser.sessionstore.restore_pinned_tabs_on_demand = true
```

I've had a bunch of google docs open in pinned tabs when I still only had 16 GB of RAM... That's when I noticed that they're eating a lot of RAM, even when I didn't even visit those tabs after restarting Firefox - that's when I learned about this property.
